Expand on config_ids. (#491)
tlswg/draft-ietf-tls-esni: draft-ietf-tls-esni.md

decreasing order of preference. This allows a server to support multiple
versions of ECH and multiple sets of ECH parameters.

## Configuration Identifiers {#config-ids}

A client-facing server has a set of known ECHConfig values, with corresponding
private keys. This set SHOULD contain the currently published values, as well as

> previous values that may still be in use. For example, clients may cache DNS
> records up to a TTL or longer.

Reviewer: ```suggestion
previous values that may still be in use, since clients may cache DNS records
up to a TTL or longer.
```

{{client-facing-server}} describes a trial decryption process for decrypting the
ClientHello. This can impact performance when the client-facing server maintains
many known ECHConfig values. To avoid this, the client-facing server SHOULD

> allocate distinct `config_id` values for each ECHConfig in its known set. The
> RECOMMENDED strategy is to, when generating an ECHConfig, select `config_id`
> randomly, and repeat if it matches any known ECHConfig.

Reviewer: ```suggestion
allocate distinct `config_id` values for each ECHConfig in its known set. The
RECOMMENDED strategy is via rejection sampling, i.e., to randomly select
`config_id` repeatedly until it does not match any known ECHConfig.
```
